edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Http/middleware/AuthenticationMiddleware.cs

```diff
@@ -63,7 +63,7 @@ public async Task Invoke(HttpContext context)
 
         internal async Task<(bool, string)> AuthenticateRequest(HttpContext context)
         {
-            if (!context.Request.Headers.TryGetValue(HttpConstants.IdHeaderKey, out StringValues clientIds) || clientIds.Count == 0)
+            if (!context.Request.Headers.TryGetValue(Constants.ServiceApiIdHeaderKey, out StringValues clientIds) || clientIds.Count == 0)
```

Comment:
I think we should do authn/authz on a per controller basis instead of in a centralized location like Middleware

Reply:
[https://msazure.visualstudio.com/One/_workitems/edit/7649811](https://msazure.visualstudio.com/One/_workitems/edit/7649811)

---
In reply to: [451721236](https://github.com/Azure/iotedge/pull/3149#discussion_r451721236) [](ancestors = 451721236)